### PR TITLE
Display node uri instead of pubkey on identity screen

### PR DESCRIPTION
**Pull request by michaelWuensch**
## Description

This PR changes the Identity screen so it shows the node uri instead of the pubkey.
A node can have more than one uri. If this is the case, the user is now able to switch between his tor and his public identity.
If only one identity is available, the switch will not be shown.

## Motivation and Context

Increase compatibility when letting other people open a channel to you using the identity screen.

## How Has This Been Tested?

My S9

## Screenshots (if appropriate):

vid

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [Contribution document](../docs/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java [hunk 1]
```diff
@@ -48,65 +50,118 @@ private void init() {
...
+
+        setupWithNodeUris(tempNodeUris, includeQRCode);
+    }
+
+    public void setupWithNodeUris(LightningNodeUri[] nodeUris, boolean includeQRCode) {
+        reset();
+        mNodeUris = nodeUris;
+        mIsQRCodeIncluded = includeQRCode;
+
+        /*
```
Comment (raphBTC) [inline app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java:65]: Can this be removed?

File: app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java [hunk 2]
```diff
@@ -48,65 +50,118 @@ private void init() {
         showAvatar();
     }
 
-    public void setupWithNodePubKey(@NonNull String nodePubKey, boolean includeQRCode) {
-        mNodePubKey = nodePubKey;
+    public void setupWithNodeUri(LightningNodeUri nodeUri, boolean includeQRCode) {
```
Comment (raphBTC) [inline app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java:53]: Is unused now

File: app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java [hunk 3]
```diff
@@ -48,65 +50,118 @@ private void init() {
...
-                if (includeQRCode) {
+    public void showTorIdentity() {
+        if (mNodeUris != null) {
+            if (mNodeUris.length > 1) {
+                for (int i = 0; i < mNodeUris.length; i++) {
+                    if (mNodeUris[i].getHost() != null && mNodeUris[i].getHost().toLowerCase().contains("onion")) {
+                        showIdentity(i);
+                        return;
+                    }
+                    showIdentity(0);
```
Comment (raphBTC) [inline app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java:104]: I'm cant quite follow the logic here. Why is this statement here in the loop?
Does this imply you always expect here to have 2 uris, where the first on has to be the public identity?

File: app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java [hunk 4]
```diff
@@ -48,65 +50,118 @@ private void init() {
...
+                }
+            } else {
+                showIdentity(0);
+            }
+        }
+    }
+
+    public void showClearNetIdentity() {
```
Comment (raphBTC) [inline app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java:112]: This looks almost exactly like `showTorIdentity`, maybe they can be merged by using an additional parameter?

File: app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java (hunk 3 above)
Comment (michaelWuensch) [reply app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java:104]: Good catch, this has of course to be outside of the for loop!

File: app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java [hunk 5]
```diff
@@ -48,65 +50,101 @@ private void init() {
...
 
-        if (mNodePubKey != null) {
-            if (!mNodePubKey.isEmpty()) {
-                if (includeQRCode) {
+    public void showIdentity(boolean tor) {
+        if (mNodeUris != null) {
+            if (mNodeUris.length > 1) {
+                for (int i = 0; i < mNodeUris.length; i++) {
+                    if (mNodeUris[i].getHost() != null ) {
+                        if(mNodeUris[i].getHost().toLowerCase().contains("onion") == tor) {
```
Comment (raphBTC) [inline app/src/main/java/zapsolutions/zap/customView/UserAvatarView.java:100]: Would probably make sense to add a `isTor()` to `LightningNodeUri` class, its used pretty often throughout this class.

**Pull request merged**